app/Livewire/Presentation/JoinAsSpeakerModal.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Livewire\Presentation;
+
+use App\Models\Presentation;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\View\View;
+use LivewireUI\Modal\ModalComponent;
+
+class JoinAsSpeakerModal extends ModalComponent
+{
+    public Presentation $presentation;
+
+    /**
+     * Initializes the component
+     * @param Presentation $presentation
+     * @return void
+     */
+    public function mount(Presentation $presentation)
+    {
+        $this->presentation = $presentation;
+    }
+
+    /**
+     * Saves the updates made on the form
+     */
+    public function save()
+    {
+        $this->authorize('joinAsCospeaker', $this->presentation);
+
+        Auth::user()->joinPresentation($this->presentation, 'speaker');
+
+        return redirect(route('presentations.show', $this->presentation))
+            ->with('status', 'Presentation successfully updated.');
+    }
```

Comment:
![image](https://github.com/HZ-HBO-ICT/it-conference/assets/54664131/223aedb8-84b6-44db-b75c-fb725fd78a03)

Don't know if this function is the cause of the error.
Clicked on the 'join as co-speaker' button, then the error occured.